core/templates/dev/head/pages/exploration_editor/editor_tab/StateSolution.js

```diff
@@ -0,0 +1,277 @@
+      $scope.correctAnswer = answer;
+    };
+
+    $scope.toggleIsActive = function() {
+      $scope.isActive = !$scope.isActive;
+    };
+
+    $scope.$on('stateEditorInitialized', function(evt, stateData) {
+      stateSolutionService.init(
+        editorContextService.getActiveStateName(),
+        stateData.interaction.solution);
+      $scope.interactionHtml = (
+        oppiaExplorationHtmlFormatterService.getInteractionHtml(
+          stateInteractionIdService.savedMemento,
+          explorationStatesService.getInteractionCustomizationArgsMemento(
+            editorContextService.getActiveStateName()),
+          'currentInteractionHtml'));
+
+      $scope.solutionIsSpecified = (
+        !!JSON.stringify(stateSolutionService.savedMemento));
+      $scope.isActive = false;
+      var interactionId = stateInteractionIdService.savedMemento;
+
+      $scope.INTERACTION_SPECS = INTERACTION_SPECS[interactionId];
+
+      if (StateSolutionHelperService.isSupportedInteraction(interactionId)) {
+        $scope.interactionHtml = null;
+        $scope.objectType = (
+          StateSolutionHelperService.getUnsupportedInteractionObjectType(
+            interactionId));
+      } else {
+        $scope.objectType = null;
+      }
+    });
+
+    $scope.getSolutionSummary = function() {
+      var solution = stateSolutionService.displayed;
+      var isExclusiveAnswer = (
+        solution.answerIsExclusive ? 'Only' : 'One');
+      var correctAnswer = StateSolutionHelperService.getCorrectAnswer(solution);
+      var explanation = (
+        $filter('convertToPlainText')(solution.explanation));
+      return (
+        '[' + isExclusiveAnswer + ' solution is ' + correctAnswer + '] ' +
+        explanation);
+    };
+
+    // This returns false if the current interaction ID is null.
+    $scope.isCurrentInteractionLinear = function() {
+      var interactionId = stateInteractionIdService.savedMemento;
+      return interactionId && INTERACTION_SPECS[interactionId].is_linear;
+    };
+
+    $scope.openAddSolutionModal = function() {
+      alertsService.clearWarnings();
+      $rootScope.$broadcast('externalSave');
+      $scope.isActive = false;
+
+      $modal.open({
+        templateUrl: 'modals/addSolution',
+        backdrop: 'static',
+        scope: $scope,
+        controller: [
+          '$scope', '$rootScope', '$modalInstance', 'editorContextService',
+          function($scope, $rootScope, $modalInstance, editorContextService) {
+            $scope.EXPLANATION_FORM_SCHEMA = {
+              type: 'html',
+              ui_config: {}
+            };
+
+            $scope.INTERACTION_SPECS = INTERACTION_SPECS;
+
+            $scope.tmpSolution = {};
+            $scope.tmpSolution.answerIsExclusive = false;
+            $scope.tmpSolution.correctAnswer = '';
+            $scope.tmpSolution.explanation = '';
+
+            $scope.addSolutionForm = {};
+
+            if ($scope.objectType === 'CodeString') {
```

Comment:
This is a really weird special case to see here. What is going on, more generally -- are you setting a default value of some sort? If so we should have some kind of registry for that so that it works automatically when new interactions are added.

In general there should be no reference to individual interaction IDs within the code in this controller, which is meant to be a general one.